rgw: civetweb update to 1.10

theanalyst opened the pull request:
wip branch for civetweb 1.10 migration

Reverted the following downstream patches and reapplied a few:
* b6ab7abac864d243e24785e5a110f418dc0174d2 - Build fix, NO_SSL_DL; seems to be present in the current civetweb.c
* f8504eff9e745c86a577f4207341577c36312f05 - introduces mg_get_local_addr: reapplied as 9aae1128b7bf9537d353891edc58b14d8f4e66a1
* 09c580e9aa11a4e75ceaebde8ae18b6e0d43bf9e: return 400 instead of 411 for invalid headers? reapplied as 302db2ddb5644ff03e51fb699a3c639a7768a1d0 cc @yehudasa 
* 3097b6975c40e7255edbe6b488a6b46642483f5c: introduces following enhancements, 
- canonicalize url path, 
- get_request_len 
- mg_set_http_status
reapplied in c598009e253cba9b8d2944faac33f28d343f9cf7

The following while reverted for easier merge; are already part of current upstream and this changeset 
* 5da61b9dffbe7e7d4667880f2ca136cb39680839 & a24ddf60bb1b0467121e26261d63da0548a4b3c1: downstream pr: https://github.com/ceph/civetweb/pull/16 which was already accepted upstream as https://github.com/civetweb/civetweb/pull/397
* b39a1f20d6f42fa0a8ef3e77e810c81044da5d02 transfer encoding takes precedence over content-length, downstream pr https://github.com/ceph/civetweb/pull/22, merged upstream as https://github.com/civetweb/civetweb/pull/511 
* e8b60cd716b05dd3556f171c976e12efc39c605d: discard unread data on read; already upstream as https://github.com/civetweb/civetweb/pull/362

mdw-at-linuxbox committed:
downstream: Add "mg_get_local_addr".  Return the local listening address used.

civetweb can listen to multiple addresses.  Some of these can have
ssl enabled, and some not.  The "using_ssl" flag is
already returned to the client, but the local address (lsa) is not,
This allows a calling application to determine on which local
address/portno a connection was accepted.

Signed-off-by: Marcus Watts <mwatts@redhat.com>
Signed-off-by: Abhishek Lekshmanan <abhishek@suse.com>
(cherry picked from commit 7f9f5d1a7dacc81e4866d5ec8cd2265c5850a0e0)

 Conflicts:
  src/civetweb.c
  whitespace like conflict due to large incoming changes in v1.10, since this is
  a very small function, it was easy to cherry-pick after the changeset

bel2125 committed:
Fix some warnings

bel2125 committed:
Disable GCC warning (flaw in GCC socket macros)

bel2125 committed:
Try to disable warning, caused by a flaw in GCC socket macros

Using select to solve #474 needs to use socket macros, but they seem
to cause sign-conversion warnings for some GCC versions.

http://www.linuxquestions.org/questions/programming-9/impossible-to-use-gcc-with-wconversion-and-standard-socket-macros-841935/

Since the socket macros are defined in the standards, an explicit
type cast must not be used - it would cause warnings for standard
compliant compilers.

bel2125 committed:
Fix copy paste error in example

bel2125 committed:
Ignore leading \r\n in multipart messages (see #470)

bel2125 committed:
Unit test: Add diagnostic message, if CGI test executable does not exist.

It would also be possible to build the test executable from a CMakeFile,
but we create it when preparing the test environment, e.g., by calling
gcc in a .travis.yml file. This way we test the build/test environment
before using the actual CMakefile or any source file.
Rarely, but nevertheless every now and then, something goes wrong when
preparing the test environment. This kind of errors are detected before
starting the CMake-build when executing .travis.yml or appveyor.yml.

If someone uses CMake on a local machine without executing the build
step for the CGI test executable manually, the test will fail.

This commit adds a diagnostic message, telling the user to create the
cgi_test.cgi executable before starting the test.
This was a remaining request from #402.

In case this kind of check is no longer required, one might still add
the build for cgi_test.cgi to the CMakeFile of the unit test.

bel2125 committed:
Fix signed/unsigned warning

bel2125 committed:
Unit test: Fix copy/paste error

bel2125 committed:
Unit test: Fix copy/paste error

bel2125 committed:
Unit test: Fix copy/paste error

bel2125 committed:
Format code

bel2125 committed:
Unit test: print content of directory, if cgi test file is missing

bel2125 committed:
Unit test: CGI test is not yet available for windows, so don't check for the CGI exe as precondition

btsimonh committed:
Add general usercontext variable to CPP server

btsimonh committed:
C++ CivetServer - Set UserContext from constructor; add getUserContext

bel2125 committed:
Merge pull request #489 from btsimonh/btsimonh-cpp-usercontext

Add general usercontext variable to CPP server

bel2125 committed:
Initialize time for socket write timeout and optimize timeout code

bel2125 committed:
Format code after merge

bel2125 committed:
Don't allow more than 2 GB of websocket payload (#488)

nigels-com committed:
Optional CMake opt-out of ASAN in debug mode build: -DCIVETWEB_ENABLE_ASAN=OFF

nigels-com committed:
CIVETWEB_ENABLE_ASAN fixup - needed in two places

bel2125 committed:
Merge pull request #490 from nigels-com/cmake-asan-optional

Optional CMake opt-out of ASAN in debug mode build

bel2125 committed:
Makefile should create all required directories

bel2125 committed:
Update some documentation

bel2125 committed:
Unit test: Avoid 'unused variable st' warning

bel2125 committed:
Format code

bel2125 committed:
Use global mutex if no atomic operations are available

bel2125 committed:
Move one time initializations in mg_init_library

bel2125 committed:
Avoid warnings for unused system functions

bel2125 committed:
GCC: Functions marked as 'unused' MAY be unused - dont raise an error if they are used

bel2125 committed:
Mark Symbian support as dead

bel2125 committed:
Add uptime information to server status (Step 21/?) (See #243)

bel2125 committed:
Remove OS process information from server status, add start time (Step 22/?) (See #243)

bel2125 committed:
Disable unused function warning for GCC

GCC unused function attribute seems fundamentally broken.
Several attempts to tell the compiler "THIS FUNCTION MAY BE USED
OR UNUSED" for individual functions failed.
Either the compiler creates an "unused-function" warning if a
function is not marked with __attribute__((unused)).
On the other hand, if the function is marked with this attribute,
but is used, the compiler raises a completely idiotic
"used-but-marked-unused" warning - and
  #pragma GCC diagnostic ignored "-Wused-but-marked-unused"
raises error: unknown option after ‘#pragma GCC diagnostic’.
Disable this warning completely, until the GCC guys sober up
again.

bel2125 committed:
Automated unit test: Set timeout for all server tests

bel2125 committed:
Test mg_set_user_connection_data with const param (see #452)

bel2125 committed:
Undo 'Test mg_set_user_connection_data with const param (see #452)'

bel2125 committed:
Reset user connection data when closing the connection

bel2125 committed:
Fix copy paste errors and add missing forward declaration

bel2125 committed:
Fix copy paste errors and add missing forward declaration

bel2125 committed:
Indent and comment clock_gettime replacement code for Windows

bel2125 committed:
Indent and comment clock_gettime replacement code for Windows

bel2125 committed:
Restructure some logic (move between functions)

bel2125 committed:
Autoformat (no manual code changes)

bel2125 committed:
Move connection init to a new function

bel2125 committed:
Add experimental interface for initializing connection specific user data

Possible solution for #452

bel2125 committed:
Add "allow gzip encoding" as flag for the request

Required, if on the fly gzip is implemented later (#199)

bel2125 committed:
Prepare for on the fly content compression (#199)

bel2125 committed:
Experimental interface to get statistics data on connections

This interface may be changed or completely removed later.
It provices additional statistics data (See #243).
This is step 1 for test purposes - more data will follow if it works.

bel2125 committed:
Fix build for NO_FILES configuration

bel2125 committed:
Add experimental connection status (see #243)

asomers committed:
Fix bash path in shebangs

"/bin/bash" is a Linuxism.  "/usr/bin/env bash" is portable.

bel2125 committed:
Merge pull request #495 from asomers/bin_bash

Fix bash path in shebangs

bel2125 committed:
Add cast to avoid warning (#496)

bel2125 committed:
Travis CI: Try 32 bit build by adding ARCH="x86" argument"

bel2125 committed:
Fix warning for 32 bit code

bel2125 committed:
Add note on proposed interface change (#498)

bel2125 committed:
Fix yet another 32 bit warning (#496)

bel2125 committed:
Fix static code analyzer warnings and format code

bel2125 committed:
Contribution.md: Ask for a description of all pull requests

bel2125 committed:
Add note on proposed pattern definition change (#499)

bel2125 committed:
Fix warning (#496)

bel2125 committed:
Try to add 32 bit CI test for Travis (see #496)

bel2125 committed:
Travis: Undo test for GCC-6

Test added for #496 causes Travis CI to fail completely,
although it follows the examples from
https://docs.travis-ci.com/user/languages/cpp/

bel2125 committed:
Add additional diagnosis data for connection states (improves #243)

bel2125 committed:
Add more connection diagnosis info (see #243)

bel2125 committed:
Fix position for USE_SERVER_STATS define

bel2125 committed:
Document experimental diagnosis interface (Final step of #243)

Further diagnosis information may be added independent of #243

bel2125 committed:
Even more statistics data (#243)

bel2125 committed:
Add Lua function to list all options

bel2125 committed:
Convert system info text to JSON format

bel2125 committed:
Docs: Add note on Redistributable version

bel2125 committed:
Windows: Show context info in dialog

bel2125 committed:
Enable server statistics in windows standalone build

bel2125 committed:
Document server statistics API - Step 22/22 (or 99/99) of #243

bel2125 committed:
Comment use of "const" for handles (mg_context*,mg_connection*)

The two opaque pointers "struct mg_context *" and "struct mg_connection *"
are used as handles. A "const" attribute for mg_context means "dont modify
the running server", while "dont modify the connection" basically means
"dont read/write data".
Add some documentation for this, also taking #452 into account.

bel2125 committed:
Update release notes

See also
git log --format=oneline b2a9ad06ab33a570af71a439e9d3e9a6c17bc68d..HEAD

For a full list since the last release, see (for 1.9 / 1.9.1):
git log --format=oneline 269e3f56256fe650d90233b8b34e667c49dda3b3..HEAD
git log --format=oneline ec7f68a2e264931cf129ec8771d2b69372dc2bfd..HEAD

bel2125 committed:
Update CREDITS

git log --format="%aN" 1cba00b2935314c186a49e49d3ac3bd5e211cbd4..HEAD | sort -u

bel2125 committed:
Update README and RELEASE_NOTES

kbranigan committed:
Update mg_websocket_write.md

Simple typo fixes, nothing significant.

bel2125 committed:
Merge pull request #500 from kbranigan/patch-1

Update mg_websocket_write.md

BigJoe01 committed:
Added background script parameters and shutdown detection

bel2125 committed:
Make ALTERNATIVE_QUEUE the new default

bel2125 committed:
Merge pull request #501 from BigJoe01/master

Added background script parameters and shutdown detection

bel2125 committed:
Fix alternative queue for systems with sizeof(int) != sizeof(void*)

bel2125 committed:
Disable unavoidable warnings

bel2125 committed:
Make mg_send_digest_access_authentication_request public

bel2125 committed:
Add missing functions to interface documentation

bel2125 committed:
Add interface for http authentication (see #426)

bel2125 committed:
Unit test: use uniform timeouts for server tests

bel2125 committed:
Rename parameter "domain" to "realm", to use the same parameter names

bel2125 committed:
Document HTTP digest auth API functions (see #426)

bel2125 committed:
Add cast for WaitForSingleObject(INFINITY), to avoid MinGW warning

bel2125 committed:
Disable warning for missing type cast to unknown type

bel2125 committed:
Unit test: add some test points

bel2125 committed:
Unit test: disable some warnings from the test framework

bel2125 committed:
Unit test: Move all test cases (test functions) into individual tests

bel2125 committed:
Unit test: disable some warnings from the test framework

bel2125 committed:
Unit test: Disable unknown pragma warning for unit tests

bel2125 committed:
Test integration for codecov

bel2125 committed:
Unit test: Try to disable all GCC warnings from the unit test framework

bel2125 committed:
Experimental fix for OSX warning (see #503)

bel2125 committed:
Unit test: Temporarily disable some unit test marks to find a timeout

bel2125 committed:
Unit test: Temporarily disable some unit test marks to find a timeout, avoid "unused" warning

bel2125 committed:
Unit test: Temporarily disable some unit test marks to find a timeout (2)

bel2125 committed:
Unit test: Format some commented code

bel2125 committed:
Unit test: Temporarily disable some unit test marks to find a timeout (3)

bel2125 committed:
Avoid const cast (see #504)

bel2125 committed:
Unit test: comment quoted string to repeat integration test

bel2125 committed:
Unit test: comment quoted string to repeat integration test

bel2125 committed:
Fix handle_form for quoted strings and reactivate unit test

kuzi117 committed:
Allow disabling executable install.

This keeps previous functionality by having executable installation on by
default. It also tries to mitigate the ability to "screw it up" by only
allowing users that have made themselves advanced to disable installing
the executable.

See: https://cmake.org/cmake/help/v3.9/command/mark_as_advanced.html

Closes #505.

bel2125 committed:
Merge pull request #506 from kuzi117/master

Allow disabling executable install.

Commit:
Background script documentation

bel2125 committed:
Fix copy/paste error in Lua reg_llstring

bel2125 committed:
Update contribution guidelines

bel2125 committed:
Update credits

git log --format="%aN" 0ec5c1d3797cebfa11715b0fa28ca646296c83b1..HEAD | sort -u

bel2125 committed:
Update release notes

    See also
    git log --format=oneline 5aaed4416f14cff5a27b475a73a9681911f8a70b..HEAD

    For a full list since the last release, see (for 1.9 / 1.9.1):
    git log --format=oneline 269e3f56256fe650d90233b8b34e667c49dda3b3..HEAD
    git log --format=oneline ec7f68a2e264931cf129ec8771d2b69372dc2bfd..HEAD

bel2125 committed:
Fix list format

Fix format of list of modifications that need to be discussed prior to merging.

bel2125 committed:
Fix some missing casts for compiling with C++ (see comment on #504)

bel2125 committed:
Add codecov badge (evaluation)

bel2125 committed:
Fix warning from static source code analysis

bel2125 committed:
Update SQLite to 3.20.0

bel2125 committed:
Update to LuaSQLite 0.9.4

bel2125 committed:
Update duktape to 1.8.0

bel2125 committed:
Modify new lsqlite3 version for CivetWeb

bel2125 committed:
Remove unreachable code

bel2125 committed:
Unit test: first step of a small https example with additional security

bel2125 committed:
Add some additional SSL security flags

bel2125 committed:
Move variable declaration to top of the function (C90 standard)

bel2125 committed:
Unit test: format code

bel2125 committed:
Add Visual Studio 2010 build for AppVeyor

bel2125 committed:
Unit-Test: Deactivate some test points

bel2125 committed:
Unit test: Reactivate test points

bel2125 committed:
Unit test: fix some copy/paste errors

bel2125 committed:
Unit test: Print all messages to stdout

bel2125 committed:
Unit test: perform minimal https server test only if NO_SSL is not set

bel2125 committed:
Move variable declaration to top of the function (C90 compatibility)

bel2125 committed:
Fix copy paste

bel2125 committed:
Fix unit test

bel2125 committed:
Unit test: fix callbacks

bel2125 committed:
Unit test: Avoid 'unused function' warning

bel2125 committed:
Unit test: test_log_message must not interfere with internal error handling

bel2125 committed:
Unit test: don't use most recent TLS settings on Travis CI build images

bel2125 committed:
Disable warning for different definitions of OpenSSL SSL_CTX_set_info_callback

bel2125 committed:
Disable SSL_CTX_set_info_callback warning only for clang (GCC does not know this pragma)

bel2125 committed:
Unit test: Try to use forked check framework

bel2125 committed:
AppVeyor: Add Visual Studio 2012 and Visual Studio 2015 unit test

bel2125 committed:
Unit test: Use a patched version of 'check' framework

bel2125 committed:
Fix syntax error in test CMakeFile

bel2125 committed:
Edit CMakeFile to flush AppVeyor build cache and force reloading check

bel2125 committed:
Edit CMakeFile to flush AppVeyor build cache and force reloading check

bel2125 committed:
Edit CMakeFile to flush AppVeyor build cache and force reloading check

bel2125 committed:
Edit CMakeFile to flush AppVeyor build cache and force reloading check

bel2125 committed:
Edit CMakeFile to flush AppVeyor build cache and force reloading check

bel2125 committed:
Edit CMakeFile to flush AppVeyor build cache and force reloading check

bel2125 committed:
Unit test: use strict C89

rzarzynski committed:
Reset mg_connection::content_len for chunked transfer encoding.

It looks that the content_len field of mg_connection isn't reset
by reset_per_request_attributes() nor in get_requests() when it
comes to chunked transfer encoding of HTTP.

Signed-off-by: Radoslaw Zarzynski <rzarzyns@redhat.com>

bel2125 committed:
Unit test for OSX: mark_point() seems to cause a sporadic hang for OSX

bel2125 committed:
Add a list of headers with / without body data and set content length

bel2125 committed:
Unit test: Add lost #endif

bel2125 committed:
Unit test: remove extra const after restoring C89/C90 compliance

bel2125 committed:
Reset content length and format code

bel2125 committed:
Disable nonsense warning -Wdate-time

bel2125 committed:
Disable nonsense warning -Wdate-time(fix)

bel2125 committed:
Disable nonsense warning -Wdate-time (3)

bel2125 committed:
Disable nonsense warning -Wdate-time (4)

bel2125 committed:
Disable nonsense warning -Wdate-time (5)

bel2125 committed:
Disable nonsense warning -Wdate-time (6)

bel2125 committed:
Disable a warning that has been renamed

bel2125 committed:
Unit test: compliance to old c standard

bel2125 committed:
Travis CI: List all available gcc versions

bel2125 committed:
Travis: Fix compiler versions to gcc-4.6 and clang-3.4

It seems Travis CI updated the build image some days ago.
The new image uses a newer compiler, and this causes spurious and
idiotic warnings, that cannot be deactivated using a pragma

e.g.,
  "expansion of date or time macro is not reproducible",
  -Wdate-time when using __DATE__

or

 "call to ‘__fread_chk_warn’ declared with attribute warning:
  fread called with bigger size * nmemb than length of destination buffer,
  return __fread_chk (__ptr, __bos0 (__ptr), __size, __n, __stream);"

or

  changed "'return' will never be executed" from -Wunreachable-code
  to -Wunreachable-code-return

None of them can be ignored in pragmas, so one can either disable -Werror
or just don't use this compiler version.

Since we are close to a 1.10 release, now it's not the time to deal
with hypochondriac compiler warnings.

On the long term, the compiler version should be fixed, and some
selected versions should compile warning free (-Werror), while others
should only compile error free.

bel2125 committed:
Try to update compilers used for Travis CI

bel2125 committed:
Travis: Try another compiler definition

bel2125 committed:
Travis: Try sudo=required, dist=precise

bel2125 committed:
Document reality: Support questions are usually a GitHub issue, not Google groups threads

For whatever reason, most people asking for help or discussing future development use
a GitHub issue, not a Google groups thread. Keep the Group for announcements, and tell
everyone issues are legitimate for questions and discussions.

bel2125 committed:
AppVeyor: Redefine set of build environments

bel2125 committed:
Unit test: fix some #ifdef and some warnings

bel2125 committed:
Lua Pages: mg.include should support absolute, relative and virtual path types

For mg.include, there should be an equivalent mechanism to SSI #include:
It should be possible to use absolute, relative and virtual path types.

See also #400

bel2125 committed:
Travis: Drop no longer required lines from .travis.yml

bel2125 committed:
Fix #ifdef error in unit test

bel2125 committed:
Drop 32 bit MinGW build (mkgmtime is missing in this environment)

bel2125 committed:
Merge branch 'master' into wip-resetconlen-in-cte

bel2125 committed:
Merge pull request #511 from rzarzynski/wip-resetconlen-in-cte

Reset mg_connection::content_len for chunked transfer encoding.
Reset to -1 for the current version.

bel2125 committed:
Increase buffer size for startup error messages

bel2125 committed:
Add HTTPS server configuration example and create new demo certificates

bel2125 committed:
Add some notes on security

bel2125 committed:
Update descriptions of examples

bel2125 committed:
AppVeyor: Store install artifacts instead of VS project files

bel2125 committed:
Document mg_check_digest_access_authentication and add an example

bel2125 committed:
AppVeyor: Copy contents of install/bin to dist

bel2125 committed:
Fix AppVeyor script and prepare for release

bel2125 committed:
Fix indent error in AppVeyor script

bel2125 committed:
Travis CI: Try to set "sudo: false" (test for IPv6)

Travis CI is shutting down the "dist: precise" build environment
https://blog.travis-ci.com/2017-08-31-trusty-as-default-status
It seems there are problems with the "sudo: required" container
when testing IPv6

bel2125 committed:
Travis CI: Test on trusty

The "precise" build environment on Travis is in the process of being decommissioned
see https://blog.travis-ci.com/2017-08-31-trusty-as-default-status
The "precise=true"+"sudo=required" environment seems to lack IPv6 support

The new dist is "trusty", but the new compiler introduce new warnings

bel2125 committed:
Show activated server stats in feature list

bel2125 committed:
Travis: Allow build with warnings (until Ubuntu trusty build environment is fully supported

bel2125 committed:
Travis: Force "ALLOW_WARNINGS" to be able to build with Ubuntu trusty environment (see #512)

bel2125 committed:
Travis CI: Try "dist: trusty" + "sudo: required"

This is the only combination not yet tested, see
https://github.com/travis-ci/travis-ci/issues/8361#issuecomment-327817604

Test it, although it does not look very promising.

bel2125 committed:
Remove copy/paste error section from documentation (#514)

bel2125 committed:
Unit test: fix some memory leaks and auto-format code

bel2125 committed:
Travis CI: Try different build environments in the same .travis.yml

Since "sudo: required" does not support IPv6 and "dist: trusty" does
not run warning free, try a combination of builds according to
https://github.com/travis-ci/travis-ci/issues/8361#issuecomment-327827096

bel2125 committed:
Makefile: add WITH_ALL=1 as shortcut

bel2125 committed:
Travis: null-change in .travis.yml to repeat the build and check build system stability

bel2125 committed:
Travis: Try to set clang version explicitly

bel2125 committed:
Travis: Try clang 3.6

bel2125 committed:
Travis: Try yet another configuration

bel2125 committed:
Travis: Rewrite build matrix in .travis.yml

bel2125 committed:
Travis: Fix build count and allow warnings everywhere

bel2125 committed:
Travis: Skip common compiler version specification

bel2125 committed:
Travis: Allow default compiler for coverage build

bel2125 committed:
Windows: Set XP compatible flags for 32 bit build

bel2125 committed:
Set error, if setting a socket to non-blocking failed

bel2125 committed:
Add some debug traces for substitute scripts (diagnosis of #519)

bel2125 committed:
More instrumentation for #519

bel2125 committed:
Code should be C89/C90 compliant, and add some CGI tests

bel2125 committed:
Do not use timeout for reading CGI responses (#519)

bel2125 committed:
Unit tests: Split functions into separate tests

bel2125 committed:
Format code

bel2125 committed:
Add configuration parameter to activate resource handling by index.cgi/lua scripts

Serving sub-resources from index.cgi is a new feature that may confusion.
See #519, in particular: see 4. from https://github.com/civetweb/civetweb/issues/519#issuecomment-329590388

Not this feature is inactive by default and must be explicitly activated

bel2125 committed:
Allow to stop websocket connectionss async from the server side (#515)

bel2125 committed:
Unit test: add some test points

bel2125 committed:
HTTP Client: Allocate memory for context and connection in one block

bel2125 committed:
Example: Fix condition (#521)

bel2125 committed:
Unit test: set defines for local debugging

bel2125 committed:
Unit test / AppVeyor: Add printf instrumentation

Try to find a segfault, that only occurs in a unit test on AppVeyor
for debug builds. It does not occur for Linux, and it does not occur
in a local debugger with Visual Studio on windows.

bel2125 committed:
AppVeyor: Try to find segfault in debug build (not reproducible in local debugger) (1/?)

bel2125 committed:
AppVeyor: Try to find segfault in debug build (not reproducible in local debugger) (2/?)

bel2125 committed:
AppVeyor: Try to find segfault in debug build (not reproducible in local debugger) (3/?)

bel2125 committed:
AppVeyor: Try to find segfault in debug build (not reproducible in local debugger) (4/?)

bel2125 committed:
AppVeyor: Try to find segfault in debug build (not reproducible in local debugger) (5/?)

bel2125 committed:
AppVeyor: Try to find segfault in debug build (not reproducible in local debugger) (6/?)

bel2125 committed:
AppVeyor: Try to find segfault in debug build (not reproducible in local debugger) (7/?)

bel2125 committed:
Restore AppVeyor build

bel2125 committed:
Add mg_ or MG_ prefix to all symbols in civetweb.h

Some symbols in civetweb.h are missing a mg_ or MG_ prefix.
Add these symbols with the new prefix now, so the symbols without prefix
can be phased out. See #522

bel2125 committed:
Release CivetWeb version 1.10

theanalyst committed:
Revert "ceph enhancements:"

This reverts commit 3097b6975c40e7255edbe6b488a6b46642483f5c.

theanalyst committed:
Revert "build fix:"

This reverts commit 0c80c0b95600ca215d247a4d90005466f518e433.

theanalyst committed:
Revert "Add "mg_get_local_addr".  Return the local listening address used."

This reverts commit 7f9f5d1a7dacc81e4866d5ec8cd2265c5850a0e0.

theanalyst committed:
Revert "make auth domain check optional via a configurable"

This reverts commit a24ddf60bb1b0467121e26261d63da0548a4b3c1.

theanalyst committed:
Revert "Allow using sub-domains in absolute URIs"

This reverts commit 5da61b9dffbe7e7d4667880f2ca136cb39680839.

theanalyst committed:
Revert "Transfer-Encoding takes precedence over Content-Length."

This reverts commit b39a1f20d6f42fa0a8ef3e77e810c81044da5d02.

theanalyst committed:
Revert "civetweb: return 400 instead of 411 on invalid input"

This reverts commit 09c580e9aa11a4e75ceaebde8ae18b6e0d43bf9e.

theanalyst committed:
Revert "discard unread data if handled by begin_request()"

This reverts commit e8b60cd716b05dd3556f171c976e12efc39c605d.

theanalyst committed:
Merge tag 'v1.10' into wip-rgw-1.10

theanalyst committed:
downstream: civetweb: reapply 3097b6975c40e7255edbe6b488a6b46642483f5c

re applies the following features as a part of the original commit atop of 1.10
canonicalize_url_path: defaults true, if false do not get rid of // ..
validate_http_method: defaults true, if false do not check for GET/PUT/etc.
  also account for validate_http_method when testing parse_http_message
get_request_len - in_content hack: be more forgiving of header charsets.
mg_set_http_status

Signed-off-by: Abhishek Lekshmanan <abhishek@suse.com>

theanalyst commented on src/civetweb.c:
```diff
@@ -5906,8 +8135,10 @@ connect_socket(struct mg_context *ctx /* may be NULL */,
 	}
 
-#ifndef NO_SSL_DL
-	if (use_ssl && (SSLv23_client_method == NULL)) {
```
earlier downstream commit b6ab7abac864d243e24785e5a110f418dc0174d2 which adds the ifndef NO_SSL_DL seems to be considered in the upstream changes

theanalyst committed:
downstream: civetweb: reapply 09c580e9aa11a4e75ceaebde8ae18b6e0d43bf9e

civetweb return 400 instead of 411 on invalid input; Doing a manual cherry-pick
as the getreq function no longer exists and is replaced by a get_request and
get_response each doing this particular check.

Signed-off-by: Abhishek Lekshmanan <abhishek@suse.com>

cbodley commented:
this is probably a good time to fix the branch naming.. once this is ready, can we add a ceph-master branch that tracks the ceph master branch?

theanalyst committed:
downstream: introduce allow_unicode_in_urls

Drop validation on urls as unicode urls are supported for things like swift
object names where client may not always send a http escaped url, swift tests like
test.functional.tests:TestFileUTF8.testQuestionMarkInName fail otherwise

Signed-off-by: Abhishek Lekshmanan <abhishek@suse.com>

cbodley commented:
@theanalyst so is this effectively the civetweb upstream branch with those two `downstream: ` commits on top?  can you just push that as a new ceph-master branch instead of merging it into wip-listen4?

theanalyst commented:
@mdw-at-linuxbox @cbodley I additionally added a downstream feature called allow_unicode_in_urls a95285e02117998638cb1e2541fec3fd7c2c23e0 as civetweb was enforcing strict url checking, now the tests seem to be green at https://github.com/ceph/ceph/pull/21123, just did a cosmetic rebase reordering the commits

theanalyst commented:
Changelog:
* fixed branch naming in civetweb:
ceph-jewel ceph-luminous and ceph-master introduced, which track the respective release branch sha1s now. Re targetted this pr against ceph-master

* reworded all the downstream commits with a downstream prefix
we have 4 downstream commits now, it is easier to revert/reapply as necessary during future upgrades
